src/Components/Artwork/Details/PrimaryLabelLine.tsx

```diff
@@ -1,27 +1,44 @@
 import { Text } from "@artsy/palette"
 import { useArtworkGridContext } from "Components/ArtworkGrid/ArtworkGridContext"
 import { SystemQueryRenderer } from "System/Relay/SystemQueryRenderer"
+import { useFulfillmentOptionsForArtwork } from "Utils/Hooks/useFulfillmentOptions"
 import type { PrimaryLabelLineQuery } from "__generated__/PrimaryLabelLineQuery.graphql"
 import type { PrimaryLabelLine_artwork$data } from "__generated__/PrimaryLabelLine_artwork.graphql"
+import type { PrimaryLabelLine_me$data } from "__generated__/PrimaryLabelLine_me.graphql"
 import { type FC, useEffect } from "react"
 import { createFragmentContainer, graphql } from "react-relay"
 
 interface PrimaryLabelLineProps {
   label: string | null | undefined
   artwork?: PrimaryLabelLine_artwork$data
+  me?: PrimaryLabelLine_me$data
+}
+
+const doesShipFree = (
+  artwork: PrimaryLabelLine_artwork$data,
+  me?: PrimaryLabelLine_me$data,
+): boolean => {
+  const result = useFulfillmentOptionsForArtwork(artwork, me)
+  const { freeShippingToUserCountry, freeGlobalShipping } = result || {}
+  return !!(freeShippingToUserCountry || freeGlobalShipping)
 }
 
 export const PrimaryLabelLine: React.FC<
   React.PropsWithChildren<PrimaryLabelLineProps>
-> = ({ label, artwork }) => {
+> = ({ label, artwork, me }) => {
   const { hideSignals, updateSignals } = useArtworkGridContext()
   const partnerOffer = artwork?.collectorSignals?.partnerOffer
+  const shipsFree = artwork && doesShipFree(artwork, me)
```

Comment:
we should check for artwork existence inside the function, which is already doing a lot of work. Return a false value from there if artwork doesn't exist.